Automattic/newspack-custom-content-migrator
# Link NYC

## Changes
- Mologui authors to CAP GA as a general reusable command `wp newspack-content-migrator molongui-to-cap` in `src/Command/General/MolonguiAutorship.php`
- Several updates to existing CAP logic `src/Logic/CoAuthorPlus.php`:
  - `create_guest_author` used to check if there was an existing GA with same `user_login` and return that one. However, it's more accurate to target by `display_name` and return that GA if it already exists
  - added `get_unique_user_login` call to `create_guest_author` command. When creating a GA, it will check if that `user_login` is available (because if there's an existing WP_User with same `user_login`, or an existing GA with same `cap-user_login` creation will fail. That shouldn't be this function's concern, because we should be able to create GAs using just `display_name` and not worry about `login` actually. And then once a GA is successfully created (with its own unique `user_login`) it can easily be linked with an existing WP_User (regardless of its login).

## How to test
- should be run on a site that has Molongui authorship. However since this is difficult to set up, I already run it in [this Asana task](https://app.asana.com/0/1200980714215013/1206116576822725/f)
- therefore I suggest just reading the code, and our TAM should be QA-ing results

---
